src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Undo the most recent command.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + "Undo the most recent command "
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_UNDO_SUCCESS = "Successfully undo the most recent command";
+    public static final String MESSAGE_UNDO_FAIL = "Cannot undo";
+
+    /**
+     * Constructs a UndoCommand to undo the most recent command.
+     */
+    public UndoCommand() {
+
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Model previousModel = model.getPreviousModel();
+        if (previousModel == null) {
+            throw new CommandException(MESSAGE_UNDO_FAIL);
+        } else {
+            model.setAddressBook(previousModel.getAddressBook());
+            model.setUserPrefs(previousModel.getUserPrefs());
+            model.setPreviousModel(previousModel.getPreviousModel());
+            return new CommandResult(MESSAGE_UNDO_SUCCESS);
+        }
+    }
```

Comment:
According to the CS2103T code quality conventions, it recommends that we extract the "happy path".
![image](https://user-images.githubusercontent.com/58943071/96348857-6f3d3b00-10de-11eb-8894-3ea2f4af1828.png)

Reply:
Yep, have updated!